boards/arm/adafruit_feather_m0_basic_proto/doc/adafruit_feather_m0_basic_proto.rst

```diff
@@ -0,0 +1,150 @@
+
+Adafruit Feather M0 Basic Proto
+###############################
+
+Overview
+********
+
+The Adafruit Feather M0 Basic Proto is a thin, light ARM development
+board with an onboard battery connector and charger for 3.7 V lithium
+polymer batteries, charging status indicator and user LEDs, native USB
+connector, 20 I/O pins, and a small prototyping area.
+
+.. image:: img/adafruit_feather_m0_basic_proto.jpg
+     :width: 640px
+     :align: center
+     :alt: Adafruit Feather M0 Basic Proto
+
+Hardware
+********
+
+- ATSAMD21G18A ARM Cortex-M0+ processor at 48 MHz
+- 32.768 kHz crystal oscillator
+- 256 KiB flash memory and 32 KiB of RAM
+- Battery connector and charger for 3.7 V lithium polymer batteries
+- Charging indicator LED
+- User LED
+- Reset button
+- Native USB port
+
+Supported Features
+==================
+
+The adafruit_feather_m0_basic_proto board configuration supports the
+following hardware features:
+
++-----------+------------+--------------------------------------+
+| Interface | Controller | Driver/Component                     |
++===========+============+======================================+
+| NVIC      | on-chip    | Nested vector interrupt controller   |
++-----------+------------+--------------------------------------+
+| Flash     | on-chip    | Can be used with NFFS to store files |
++-----------+------------+--------------------------------------+
+| SYSTICK   | on-chip    | Systick                              |
++-----------+------------+--------------------------------------+
+| WDT       | on-chip    | Watchdog                             |
++-----------+------------+--------------------------------------+
+| GPIO      | on-chip    | I/O ports                            |
++-----------+------------+--------------------------------------+
+| USART     | on-chip    | Serial port                          |
++-----------+------------+--------------------------------------+
+| SPI       | on-chip    | Serial Peripheral Interface port     |
++-----------+------------+--------------------------------------+
+
+Other hardware features are not currently supported by Zephyr.
+
+The default configuration can be found in the Kconfig file
+:file:`boards/arm/adafruit_feather_m0_basic_proto/adafruit_feather_m0_basic_proto_defconfig`.
+
+Connections and IOs
+===================
+
+The `Adafruit Feather M0 Basic Proto Learn site`_ has detailed
+information about the board including `pinouts`_ and the `schematic`_.
+
+System Clock
+============
+
+The SAMD21 MCU is configured to use the 32.768 kHz external oscillator
+with the on-chip PLL generating the 48 MHz system clock.
+
+Serial Port
+===========
+
+The SAMD21 MCU has 6 SERCOM based USARTs.  On the Adafruit Feather M0
+Basic Proto, SERCOM0 is the Zephyr console and is available on pins 0
+(RX) and 1 (TX).
+
+SPI Port
+========
+
+The SAMD21 MCU has 6 SERCOM based SPIs.  On the Adafruit Feather M0
+Basic Proto, SPI4 is available on pin 22 (MISO), pin 23 (MOSI), and
```

Comment:
Are these pins correct?  The schematic shows SERCOM4.2/MOSI to pin 5.

Reply:
@nzmichaelh Thank you. Adafruit are using a confusing number of different pin numbering schemes on these boards (as seen on the graphic on https://learn.adafruit.com/adafruit-feather-m0-basic-proto/pinouts).

I have used the pin numbers used in the official Adafruit board documentation (scroll past the graphic on the above link for the list).